probe-rs/src/architecture/arm/sequences/nxp.rs

```diff
@@ -555,18 +559,225 @@ impl ArmDebugSequence for MIMXRT11xx {
+        interface.write_word_32(0x40102288, 0x1000)?;
+
+        std::thread::sleep(Duration::from_millis(100));
+
+        interface.write_word_32(0x40102208, 0x1000)?;
+
+        Ok(())
+    }
+
+    fn wait_for_stop_after_reset(&self, memory: &mut dyn ArmProbe) -> Result<(), ArmError> {
+        tracing::trace!("MIMXRT6xx::wait_for_stop_after_reset");
+
+        let ap: MemoryAp = memory.ap();
+        let dp = ap.ap_address().dp;
+
+        let dhcsr = armv8m::Dhcsr(0xA05F0003);
+
+        let start = Instant::now();
+
+        while !MIMXRT6xx::csw_debug_status(memory.get_arm_communication_interface()?, ap)?
+            && start.elapsed() < Duration::from_micros(300_000)
+        {
+            // Wait for either condition
+        }
+
+        self.enable_debug_mailbox(memory.get_arm_communication_interface()?, dp, ap)?;
+
+        // Halt the core in case it didn't stop at a breakpiont.
+        memory.write_word_32(armv8m::Dhcsr::get_mmio_address(), dhcsr.into())?;
+
+        // Clear watch point
+        memory.write_word_32(0xE0001020, 0x0)?;
+        memory.write_word_32(0xE0001028, 0x0)?;
+
+        Ok(())
+    }
+}
+
+impl ArmDebugSequence for MIMXRT6xx {
+    fn debug_port_start(
+        &self,
+        interface: &mut ArmCommunicationInterface<Initialized>,
+        dp: DpAddress,
+    ) -> Result<(), ArmError> {
+        tracing::trace!("MIMXRT6xx::debug_port_start");
+
+        // This is not mentioned in the PDSC
+        // If the errors aren't cleared before starting this will fail
+        MIMXRT6xx::clear_errors(interface, dp)?;
+
+        debug_port_start(interface, dp, Select(0))?;
+
+        let ap = ApAddress { dp, ap: 2 };
+        let ap = MemoryAp::new(ap);
+
+        self.enable_debug_mailbox(interface, dp, ap)?;
+
+        Ok(())
+    }
+
+    fn reset_system(
+        &self,
+        interface: &mut dyn ArmProbe,
+        _: probe_rs_target::CoreType,
+        _: Option<u64>,
+    ) -> Result<(), ArmError> {
+        tracing::trace!("MIMXRT6xx::reset_system");
+
+        //Halt the core
+        let dhcsr = armv8m::Dhcsr(0xA05F0003);
+        interface.write_word_32(armv8m::Dhcsr::get_mmio_address(), dhcsr.into())?;
+
+        //clear VECTOR CATCH and set TRCENA
+        let tmp = interface.read_word_32(armv8m::Demcr::get_mmio_address())?;
+        let mut demcr = armv8m::Demcr(tmp);
+        demcr.set_trcena(true);
+        demcr.set_vc_corereset(false);
+        interface.write_word_32(armv8m::Demcr::get_mmio_address(), demcr.into())?;
+
+        self.reset_flash(interface)?;
+
+        // Set watch point at SYSTEM_STICK_CALIB access
+        interface.write_word_32(0xE0001020, 0x50002034)?;
+        interface.write_word_32(0xE0001028, 0x00000814)?;
+
+        let mut aircr = armv8m::Aircr(0);
+        aircr.vectkey();
+        aircr.set_sysresetreq(true);
+
+        // __errorcontrol = 1;
+        // This means that errors should be ignored in this region
+        #[allow(unused_must_use)]
```

Comment:
Please don't use this, if you have to ignore a return value use `let _ = xxx;`.

Reply:
Will do. Would you mind explaining why you prefer that way to the scoped way of doing it?

Reply:
Nvm, after looking at the code again I see that the way it was done was problematic